### ISSUE TITLE
request: Fail on trailing slash?

### ISSUE THREAD

**Issue by tolmasky** [micromatch/micromatch#115]
In the context I'm using micromatch, the expectation is that you are almost only ever wanting to match files, and matching directories is a very rare occurrence. Additionally, I know for a fact that all the directory paths I'll be passing micromatch end in a /.

Currently micromatch seems to match trailing slashes by default unless you explicitly state you don't want one: `/a/b(!/)`. I would very much like the opposite behavior, for `**/*` to *fail* on `/a/b/`, and for the user to have to manually allow the slash if they want to allow a directory: `**/*/`.

Is there any way to do this without having to manipulate the user's expression, which can get quite tricky since its not as simple as if .charAt(last) !== "/" then append `!(/)` since it might be in parenthesis, etc.

To summarize, I'm looking for a mode where "**/*" does not match items with a trailing slash, and requires the user to explicitly do `**/*/` or `**/*{,/}`.

**Comment by jonschlinkert** [comment]
This follows bash behavior, what you're requesting differs from bash.

Try this:

```js
const files = ['a/b', 'c/d/', 'e/f', 'a', 'b', 'c/']

console.log(mm(files, '**/*', { ignore: ['**/'] }));
//=> [ 'a/b', 'e/f', 'a', 'b' ]
```

You could just push an ignore pattern onto the `options.ignore` array.

**Comment by jonschlinkert** [comment]
closing since this isn't a bug and we're going to try to adhere to bash behavior.

**Comment by dawgwe1** [comment]
Close all accounts

**Comment by jonschlinkert** [comment]
@dawgwe1 you've been blocked
